index.js

```diff
@@ -55,64 +33,138 @@ app.post('/new-message', (req, res) => {
     ref.once('value').then(snapshot => {
       const chat = snapshot.val();
       const usernames = chat && chat.usernames ? chat.usernames : '';
-      console.log('le usernames', chat, usernames);
+
       if (chat && chat.message_id && chat.message_id > message.message_id) {
         console.log('discarting old message', chat.message_id, message.message_id);
-        return res.end('Error: message is undefined');
+        return res.end('discarting old message', chat.message_id, message.message_id);
       }
 
       if (messageManager.isCommand(message.entities)) {
         if (messageManager.isSet(message.text)) {
-          ref.update({
-            'message_id': message.message_id,
-            'usernames': '' + messageManager.extractUniqueUsernames(message, usernames)
-          }).then(m => {
-            talkToBot(chatId, 'Usernames added.', res);
-            console.log('Updated message:', m);
-            res.end('Updated message:', m);
-          }).catch(e => {
-            console.log('Error updating db:', e);
-            res.end('Error updating db:', e);
-          });
+          setUsernames(message, chatId, usernames, ref, res);
         } else if (messageManager.isPing(message.text)) {
-          talkToBot(chatId, usernames || 'No usernames added.', res);
+          pingAll(chatId, usernames || 'No usernames added. Use set command to add them.', res);
         } else if (messageManager.isClear(message.text)) {
-          ref.set({
-            usernames: ''
-          });
-          talkToBot(chatId, 'All cleared.', res);
+          clearUsernames(message, chatId, ref, res);
         }
       } else if (message.new_chat_participant || message.left_chat_participant) {
-        ref.set({
-          'usernames': messageManager.extractUniqueUsernames(message, usernames)
-        });
-
-        if(message.new_chat_participant) {
-          talkToBot(chatId, `${message.new_chat_participant.username} was added to the list.`, res);
-        }
-        if(message.left_chat_participant) {
-          talkToBot(chatId, `${message.left_chat_participant.username} was removed from the list.`, res);
-        }
+        addOrRemoveParticipant(message, chatId, usernames, ref, res);
       }
-      res.end('usernames', usernames);
-    }).catch(error => {
+    }, error => {
       console.log('Error reading db:', error);
       res.end('Error reading db:', error);
     });
   } catch (err) {
-    console.log('Error somewhere: ', err);
-    talkToBot(chatId, `Sorry we didn't get that`, res);
-    res.end('Error somewhere: ', err);
-  }
-
-  res.end('All done!');
+    telegramManager.talkToBot(chatId, `Sorry we didn't get that`).then(() => {
+      console.log('Error somewhere: ', err);
+      res.end('Error somewhere: ', err);
```

Comment:
Do we really need to log here? or remove `then` and `err` here and log, anyways you have to log the error on L57